libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -809,6 +809,77 @@ int32_t SOCKETS_SetSockOpt( Socket_t xSocket,
 
             break;
 
+        case SOCKETS_SO_TCPKEEPALIVE:
+           {
+               int keep_alive = *( ( int * ) pvOptionValue );
```

Comment:
You should validate `pvOptionValue` here before dereferencing it, e.g.:

```c
if( NULL == pvOptionValue )
{
    return SOCKETS_EINVAL;
}
```

Reply:
The validation of pvOptionValue is done `lwip_setsockopt()` API against NULL, hence this change is not needed